lib/onesignal/client.rb

```diff
@@ -35,6 +35,12 @@ def fetch_player player_id
       get "players/#{player_id}"
     end
 
+    def csv_export params
```

Comment:
I think it is better to use keyword arguments, as they provide a stronger contract than plain hashes while keeping the same public API

```suggestion
    def csv_export extra_fields: nil, last_active_since: nil, segment_name: nil 
      post "players/csv_export?app_id=#{@app_id}", extra_fields: extra_fields, last_active_since: last_active_since&.to_i&.to_s, segment_name: segment_name
```

Reply:
Yeah – I opted for a hash because of the normalization needed in the client. Building the hash of body params is cumbersome with keyword arguments. But I can change it if you like.

Reply:
Well if there is a benefit in using a hash I don't mind at all. Why do you say it is cumbersome compared to keyword args? It should basically be the same thing, just with some syntactic sugar.

Reply:
They are all optional and the request should silently succeed whether they are passed or not. Keyword args with a `nil` default doesn't add much value.

It's okay to use keyword args, it just results in something like:

```ruby
params = {}
params[:extra_fields] = extra_fields if extra_fields
params[:last_active_since] = last_active_since.to_i.to_s if last_active_since
params[:segment_name] = segment_name if segment_name
post "players/csv_export?app_id=#{@app_id}", params
```

which is more explicit but imho a bit annoying. This was my code originally before I went back to the hash. I don't have a strong opinion though, I'm happy either way.

Reply:
My main reason for keyword args is ignoring any key that is not defined. Basically what you did with `params.slice!` but embedded in the language syntax. This avoids receiving unexpected args.

Reply:
Cool, no problem, I will change now.